Adding Blosc2 HDF5 filter

t20100 opened the pull request:
Very early integration of blosc-2 HDF5 compression filter in hdf5plugin.
There is still some work to do regarding compilation options, enabling SIMD, using zlib-ng, and testing.

~~Form what I tested, there is an issue with zlib.~~ (fixed)

related to #194

FrancescAlted commented:
Sounds good.  I've added a couple of notes about your experience with the optimal number of threads.

t20100 commented on src/PyTables/hdf5-blosc2/src/blosc2_filter.c:
```diff
@@ -0,0 +1,304 @@
+    /* Use 4 threads for the threads.  In the future it should be allowed to
+     * set this from the public API.
+     */
+    cparams.nthreads = 4;
```
IMO 1 is a good default (considering the multiprocessing scenario), but that should be configurable by the user.
I gave some feedbacks in the original PR: https://github.com/PyTables/PyTables/pull/969#issuecomment-1329347404

t20100 commented:
There is an issue with `universal2` build on macos (see https://gitlab.esrf.fr/silx/bob/hdf5plugin/-/jobs/512995).
I'll look more at it next week

attn @FrancescAlted

t20100 commented:
Fixed issue with `zlib` (5264e11) and added P9 altivec SIMD support (1f9a9e8).

Still need to fix `universal2` build.

Remaining improvements that I would prefer to do in other PRs:
- Use zlib-ng #245 
-  Add Neon instruction support #244

t20100 commented:
Related `c-blosc2` issue: https://github.com/Blosc/c-blosc2/issues/430

t20100 commented:
This PR https://github.com/Blosc/c-blosc2/pull/431 proposes a fix to `c-blosc2` regarding `universal2` build on macos and it is included in this PR (commit [97b7b38](https://github.com/silx-kit/hdf5plugin/pull/201/commits/97b7b382c59f85cf82e4b82cce8ebcb1258a9a3d)).

To me, this can be reviewed and merged to move forward, but I'd wait for the issue to be fixed upstream before making a release of `hdf5plugin`.

Anyway, ready for review.

t20100 commented:
PR https://github.com/Blosc/c-blosc2/pull/431 has been merged.
So to me, this PR can be merged (and eventually released).

vasole reviewed (approved):
LGTM
